File: app/src/main/java/com/divercity/app/features/profile/selectgender/SelectGenderFragment.java
package com.divercity.app.features.profile.selectgender;


import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;

import com.divercity.app.R;
import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.core.utils.Preconditions;
import com.divercity.app.databinding.FragmentSelectGenderBinding;

import javax.inject.Inject;

import dagger.android.support.AndroidSupportInjection;

public class SelectGenderFragment extends BaseFragment {

    private static final int PROGRESS = 0;

    SelectGenderViewModel viewModel;
    FragmentSelectGenderBinding binding;

    @Inject
    Context context;

    public static SelectGenderFragment newInstance() {
        SelectGenderFragment fragment = new SelectGenderFragment();
        return fragment;
    }

    public SelectGenderFragment() {
        // Required empty public constructor
    }

    @Override
    public void onAttach(Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_select_gender;
    }

    @Override
    public ViewModel getViewModel() {
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(SelectGenderViewModel.class);
        return viewModel;
    }

    @Override
    public int getBindingVariable() {
        return 0;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding = (FragmentSelectGenderBinding) getViewDataBinding();
        binding.listGender.setAdapter(new SelectGenderAdapter(getContext(), listener));
        setupHeader();
        subscribeToLiveData();
    }

    private void setupHeader(){
        binding.includeHeader.progressBar.setMax(100);
        binding.includeHeader.progressBar.setProgress(0);
        binding.includeHeader.progressBar.setProgressWithAnim(PROGRESS);
        binding.includeHeader.txtProgress.setText(PROGRESS + "%");
        binding.includeHeader.txtTitle.setText(R.string.select_your_gender);

        binding.includeHeader.btnClose.setOnClickListener(view1 -> {

        });

        binding.includeHeader.btnSkip.setOnClickListener(view1 -> {

        });
    }

    private void subscribeToLiveData() {
        viewModel.dataUpdateUser.observe(this, response -> {
            Preconditions.checkNotNull(response);
            binding.includeLoading.setResource(response);
            switch (response.status){
                case ERROR:
                    break;
                case SUCCESS:
                    break;
            }
        });
    }

    SelectGenderAdapter.GenderAdapterListener listener = userType -> {
        navigator.navigateToNextProfile(getActivity(), viewModel.getAccountType());
    };
}

File: app/src/main/java/com/divercity/app/features/onboarding/selectgender/SelectGenderFragment.kt
package com.divercity.app.features.onboarding.selectgender

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.divercity.app.R
import com.divercity.app.core.base.BaseFragment
import com.divercity.app.data.Status
import kotlinx.android.synthetic.main.fragment_onboarding_header_search_list.*
import kotlinx.android.synthetic.main.view_header_profile.*
import javax.inject.Inject

class SelectGenderFragment : BaseFragment() {

    lateinit var viewModel: SelectGenderViewModel

    @Inject
    lateinit var adapter: SelectGenderAdapter

    var currentProgress: Int = 0

    companion object {
        private const val PARAM_PROGRESS = "paramProgress"

        fun newInstance(progress: Int): SelectGenderFragment {
            val fragment = SelectGenderFragment()
            val arguments = Bundle()
            arguments.putInt(PARAM_PROGRESS, progress)
            fragment.arguments = arguments
            return fragment
        }
    }

    override fun layoutId(): Int = R.layout.fragment_onboarding_header_search_list

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProviders.of(this, viewModelFactory)[SelectGenderViewModel::class.java]
        currentProgress = arguments?.getInt(PARAM_PROGRESS) ?: 0
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btn_continue.visibility = View.GONE
        include_search.visibility = View.GONE
        adapter.setListener(listener)
        list.adapter = adapter
        setupHeader()
        subscribeToLiveData()
    }

    private fun setupHeader() {
        include_header.apply {

            progress_bar.apply {
                max = 100
                progress = 0
                setProgressWithAnim(currentProgress)
            }
            txt_title.setText(R.string.select_your_gender)

            txt_progress.text = currentProgress.toString().plus("%")

            btn_close.setOnClickListener {
                navigator.navigateToHomeActivity(activity!!)
            }

            btn_skip.setOnClickListener {
                navigator.navigateToNextOnboarding(activity!!,
                        viewModel.accountType,
                        currentProgress,
                        false
                )
            }
        }
    }

    private fun subscribeToLiveData() {
        viewModel.updateUserProfileResponse.observe(this, Observer { response ->
            when (response?.status) {
                Status.LOADING -> showProgress()

                Status.ERROR -> {
                    hideProgress()
                    Toast.makeText(activity, response.message, Toast.LENGTH_SHORT).show()
                }
                Status.SUCCESS -> {
                    hideProgress()
                    navigator.navigateToNextOnboarding(activity!!,
                            viewModel.accountType,
                            currentProgress,
                            true
                    )
                }
            }
        })
    }

    private val listener: SelectGenderAdapter.Listener = SelectGenderAdapter.Listener {
        viewModel.updateUserProfile(getString(it.textId))
    }
}